lib/adapter.js

```diff
@@ -3633,15 +3633,14 @@ function Adapter(options) {
             }
 
             adapterObjects.getObjectView('system', 'device', {startkey: this.namespace + '.', endkey: this.namespace + '.\u9999'}, options, (err, obj) => {
-                if (!err && obj && obj.rows && obj.rows.length) {
-                    const res = [];
-                    for (let i = 0; i < obj.rows.length; i++) {
-                        res.push(obj.rows[i].value);
-                    }
-                    callback(null, res);
-                } else {
-                    callback(err, []);
+                if (err || !obj || !obj.rows || !obj.rows.length) {
+                    return void callback(err, []);
```

Comment:
I think we misunderstood each other. I would only short-circuit here if `err` is truthy. If any of the other conditions is true, the resulting array will be empty anyways. 

Also the way it is written now, it is likely to introduce bugs in user code where the check is backwards, e.g. like this:
```js
(err, list) => {
    if (list) { /* do stuff */ } // this will now ALWAYS be executed.
    else { /* print err */ }
}
```

Reply:
ahhh ok, I see you... fixed